### ISSUE TITLE
[Transforms] Namespace with same name as default export uses exported reference instead of local

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#8288]
**Expected**:

```js
//// [defaultExportsCannotMerge04.ts]

export default function Foo() {
}

namespace Foo {
    export var x;
}

interface Foo {
}

export interface Foo {
}

//// [defaultExportsCannotMerge04.js]
"use strict";
function Foo() {
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = Foo;
var Foo;
(function (Foo) {
})(Foo || (Foo = {}));
```

**Actual**:

```js
//// [defaultExportsCannotMerge04.ts]

export default function Foo() {
}

namespace Foo {
    export var x;
}

interface Foo {
}

export interface Foo {
}

//// [defaultExportsCannotMerge04.js]
"use strict";
function Foo() {
}
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = exports.Foo;
var Foo;
(function (Foo) {
})(exports.Foo || (exports.Foo = {}));
```

Failed tests:
*  defaultExportsCannotMerge04.ts

**Comment by mhegazy** [comment]
This is already an error condition, and an error was reported. the output seems correct given that.

**Comment by RyanCavanaugh** [comment]
No complaints it seems
